src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -2084,8 +2091,14 @@ bool isRefEscape
                                         _compilation.IsPeVerifyCompatEnabled,
                                         stackLocalsOpt: null))
                     {
+#if DEBUG
+                        CheckVisited(receiver);
```

Comment:
nit: consider marking the method itself with Conditional attribute (and if we do that maybe the method name could be updated to "AssertVisited").

Reply:
I've renamed to `AssertVisited()`. I think I'd prefer to leave as conditionally compiled though, so it's clear at the call-sites that this does not affect release builds.

Reply:
That's okay, but if the method is called "Assert..." then I think it's clear that it's a conditional method. Up to you...